src/dmd/func.d

```diff
@@ -2540,6 +2540,7 @@ extern (C++) class FuncDeclaration : Declaration
         {
             tf = new TypeFunction(ParameterList(fparams), treturn, LINK.c, stc);
             fd = new FuncDeclaration(Loc.initial, Loc.initial, id, STC.static_, tf);
+            fd.generated = true;
```

Comment:
this change seems unrelated, can you explain ?

Reply:
This changes `fd.kind()` to `generated function` which is more informative when `@nogc` inference fails due to a Druntime lowering.

Not sure if I'll keep this around because the check now recognises `foreach`-lowerings.